src/mutation.rs

```diff
@@ -0,0 +1,225 @@
+use crate::order::OrderStatus;
+use crate::order_item::OrderItem;
+use crate::query::query_object;
+use crate::user::User;
+use crate::{order::Order, query::query_order};
+
+/// Describes GraphQL order mutations.
+pub struct Mutation;
+
+#[Object]
+impl Mutation {
+    /// Creates an order with `OrderStatus::Pending`.
+    async fn create_order<'a>(
+        &self,
+        ctx: &Context<'a>,
+        #[graphql(desc = "CreateOrderInput")] input: CreateOrderInput,
+    ) -> Result<Order> {
+        authenticate_user(&ctx, input.user_id)?;
+        let db_client = ctx.data_unchecked::<Database>();
+        let collection: Collection<Order> = db_client.collection::<Order>("orders");
+        validate_input(db_client, &input).await?;
+        let current_timestamp = DateTime::now();
+        let internal_order_items = input
+            .order_items
+            .iter()
+            .map(|i| OrderItem::new(i, current_timestamp))
+            .collect();
+        let order = Order {
+            _id: Uuid::new(),
+            user: User { _id: input.user_id },
+            created_at: current_timestamp,
+            order_status: OrderStatus::Pending,
+            placed_at: None,
+            rejection_reason: None,
+            internal_order_items,
+        };
+        match collection.insert_one(order, None).await {
+            Ok(result) => {
+                let id = uuid_from_bson(result.inserted_id)?;
+                query_order(&collection, id).await
+            }
+            Err(_) => Err(Error::new("Adding order failed in MongoDB.")),
+        }
+    }
+
+    /// Places an existing order by changing its status to `OrderStatus::Placed`.
+    async fn place_order<'a>(
+        &self,
+        ctx: &Context<'a>,
+        #[graphql(desc = "Uuid of order to place")] id: Uuid,
+    ) -> Result<Order> {
+        let db_client = ctx.data_unchecked::<Database>();
+        let collection: Collection<Order> = db_client.collection::<Order>("orders");
+        let order = query_order(&collection, id).await?;
+        authenticate_user(&ctx, order.user._id)?;
+        set_status_placed(&collection, id).await?;
+        query_order(&collection, id).await
+    }
+}
+
+/// Extracts UUID from Bson.
+///
+/// Adding a order returns a UUID in a Bson document. This function helps to extract the UUID.
+fn uuid_from_bson(bson: Bson) -> Result<Uuid> {
+    match bson {
+        Bson::Binary(id) => Ok(id.to_uuid()?),
+        _ => {
+            let message = format!(
+                "Returned id: `{}` needs to be a Binary in order to be parsed as a Uuid",
+                bson
+            );
+            Err(Error::new(message))
+        }
+    }
+}
+
+/// Sets the status of an order to `OrderStatus::Placed`.
+///
+/// * `collection` - MongoDB collection to update.
+/// * `input` - `UpdateOrderInput`.
+async fn set_status_placed(collection: &Collection<Order>, id: Uuid) -> Result<()> {
+    let current_timestamp = DateTime::now();
```

Comment:
I thought we decided on the "you only have 1h time to place" thing
am I do blind to find the logic for this?